fudy/NoteScreen/StickNoteModel.cpp

```diff
@@ -0,0 +1,199 @@
+#include "StickNoteModel.h"
+
+#include <QDebug>
+
+StickNoteModel::StickNoteModel(QObject* parent)
+    : QAbstractListModel{parent}
+{
+	QSqlQuery query;
+	query.exec("SELECT * FROM fudy");
+
+	while(query.next())
+	{
+		StickItem item;
+		item.id = query.value("id").toInt();
+		item.enable = query.value("enable").toBool();
+		item.header = query.value("header").toString();
+		item.noteText = query.value("noteText").toString();
+		item.visible = query.value("visible").toBool();
+		item.xaxis = query.value("xaxis").toInt();
+		item.yaxis = query.value("yaxis").toInt();
+
+		// Thêm dữ liệu vào QVector
+		m_items.append(item);
+	}
+}
+
+int StickNoteModel::rowCount(const QModelIndex& parent) const
+{
+	if(parent.isValid())
+	{
+		return 0;
+	}
+	return m_items.size();
+}
+
+QVariant StickNoteModel::data(const QModelIndex& index, int role) const
+{
+	if(!index.isValid())
+	{
+		return QVariant();
+	}
+
+	const StickItem item = m_items.at(index.row());
+	switch(role)
+	{
+	case IdRole:
+		return QVariant(item.id);
+		break;
+	case EnableRole:
+		return QVariant(item.enable);
+		break;
+	case HeaderRole:
+		return QVariant(item.header);
+		break;
+	case NoteTextRole:
+		return QVariant(item.noteText);
+		break;
+	case Visible:
+		return QVariant(item.visible);
+		break;
+	case XAxis:
+		return QVariant(item.xaxis);
+		break;
+	case YAxis:
+		return QVariant(item.yaxis);
+		break;
+	}
+	return QVariant();
+}
+
+QHash<int, QByteArray> StickNoteModel::roleNames() const
+{
+	QHash<int, QByteArray> roles;
+	roles[IdRole] = "id";
+	roles[EnableRole] = "enable";
+	roles[HeaderRole] = "header";
+	roles[NoteTextRole] = "noteText";
+	roles[Visible] = "visible";
+	roles[XAxis] = "xaxis";
+	roles[YAxis] = "yaxis";
+	return roles;
+}
+
+bool StickNoteModel::removeItemAt(int index)
+{
+	QSqlQuery query;
+	query.prepare("delete from fudy where id = ?");
+	query.addBindValue(m_items[index].id);
+
+	beginRemoveRows(QModelIndex(), index, index);
+
+	m_items.removeAt(index);
+
+	endRemoveRows();
+
+	return query.exec();
+}
+
+bool StickNoteModel::appendNewItem()
+{
+	beginInsertRows(QModelIndex(), rowCount(), rowCount());
+
+	StickItem item;
+	item.enable = false;
+	item.header = "Task to do";
+	item.noteText = "";
+	item.visible = false;
+	item.xaxis = 0;
+	item.yaxis = 0;
+	m_items.append(item);
+
+	endInsertRows();
+
+	QSqlQuery query;
+	query.prepare(
+	    "insert into fudy (enable,header,noteText,visible,xaxis,yaxis) values (?,?,?,?,?,?)");
+	query.addBindValue(item.enable ? 1 : 0);
+	query.addBindValue(item.header);
+	query.addBindValue(NULL);
+	query.addBindValue(item.visible ? 1 : 0);
+	query.addBindValue(item.xaxis);
+	query.addBindValue(item.yaxis);
+
+	return query.exec();
+}
+
+bool StickNoteModel::removeCompletedItems()
+{
+	beginResetModel();
+	m_items.clear();
+	endResetModel();
+
+	QSqlQuery query;
+	query.prepare("DELETE FROM fudy");
```

Comment:
I dont understand this query

Reply:
Instead of delete each elements one by one, I delete the whole table.